src/util/expr_iterator.h

```diff
@@ -42,22 +42,18 @@ class const_unique_depth_iteratort;
 struct depth_iterator_expr_statet final
 {
   typedef exprt::operandst::const_iterator operands_iteratort;
-  inline depth_iterator_expr_statet(
-    const exprt &expr,
-    operands_iteratort it,
-    operands_iteratort end):
-    expr(expr), it(it), end(end) { }
+  inline depth_iterator_expr_statet(const exprt &expr) : expr(expr), op_idx(0)
```

Comment:
Missing `explicit` (and you can get rid of the `inline`, which is actually implicit...).